src/server/src/main/java/io/cassandrareaper/service/SegmentRunner.java

```diff
@@ -292,9 +273,7 @@ private boolean runRepair() {
       String keyspace = repairUnit.getKeyspaceName();
       boolean fullRepair = !repairUnit.getIncrementalRepair();
 
-      LazyInitializer<Set<String>> busyHosts = new BusyHostsInitializer(cluster);
-
-      if (!canRepair(segment, keyspace, coordinator, cluster, busyHosts)) {
+      if (!canRepair(segment)) {
```

Comment:
```suggestion
      if (RepairSegment.State.NOT_STARTED != segment.getState()) {
```

and move it up above line 273. and move the two remote calls for cluster and coordinator down… 
e.g.
```
  private boolean runRepair() {
    LOG.debug("Run repair for segment #{}", segmentId);
    RepairSegment segment = context.storage.getRepairSegment(repairRunner.getRepairRunId(), segmentId).get();
    Thread.currentThread().setName(clusterName + ":" + segment.getRunId() + ":" + segmentId);

    try (Timer.Context cxt = context.metricRegistry.timer(metricNameForRunRepair(segment)).time()) {

      if (SEGMENT_RUNNERS.containsKey(segmentId)) {
        LOG.error("SegmentRunner already exists for segment with ID: {}", segmentId);
        throw new ReaperException("SegmentRunner already exists for segment with ID: " + segmentId);
      }

      if (RepairSegment.State.NOT_STARTED != segment.getState()) {
        LOG.info(
            "Cannot run segment {} for repair {} at the moment. Will try again later", segmentId, segment.getRunId());

        try {
          Thread.sleep(SLEEP_TIME_AFTER_POSTPONE_IN_MS);
        } catch (InterruptedException ignore) { }
        return false;
      }

      Cluster cluster = context.storage.getCluster(clusterName);
      JmxProxy coordinator = clusterFacade.connect(cluster, potentialCoordinators);
      String keyspace = repairUnit.getKeyspaceName();
      boolean fullRepair = !repairUnit.getIncrementalRepair();
     …
```

Reply:
sure thing, great suggestion